src/comparisons/utils/deep_extend/modern.js

```diff
@@ -7,17 +7,19 @@ function deepExtend(out, ...arguments_) {
     if (!obj) {
       continue;
     }
-  
+
     for (const [key, value] of Object.entries(obj)) {
-        if (typeof value === 'object' && value !== null) {
-          out[key] = Array.isArray(value) ? [...value] : deepExtend(out[key], value);
-        } else {
-          out[key] = value;
-        }
+      if (typeof value === 'object' && value !== null) {
```

Comment:
Out of scope for this PR but I noticed that this code doesn't properly handle objects like `Date` objects or copying items from _within_ the array. I think #144 already chats about it